src/collision.cpp

```diff
@@ -454,8 +443,7 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 			// Find nearest collision of the two boxes (raytracing-like)
 			f32 dtime_tmp = nearest_dtime;
 			CollisionAxis collided = axisAlignedCollision(box_info.box,
```

Comment:
Thought: This is still some kind of linear approximation of a quadratic function, just now with the average speed, isn't it?

We could make `axisAlignedCollision` take acceleration into account properly: If acceleration on an axis is zero, the current code is fine. Otherwise, exactly solve the quadratic equation for the first point in time >= 0 where the faces touch. (We might want doubles here?)

So for example if we're checking for an X axis collision, we look at the current velocity. Say the X component is > 0. So we want to check whether our +X face smashes into another the -X face of another box. Now we have the distance between those faces and we want to find the point in time at which they touch; we then check whether the YZ projections of the AABBs overlap. That's what the code is essentially currently doing for all three axes, except it doesn't take acceleration into account: It's just `dtime = distance / std::abs(speed.Y);` when it should be $vt + 1/2at = d$ where $d$ is the distance between the faces solved for $t$.

Reply:
I did not previously check `axisAlignedCollision`, and this fix here is quite *urgent* IMHO - there are plenty of complaints in minetest games such as Voxelibre and Mineclonia about mobs bouncing (on beds) through roofs.
Hence I opened #15236 to discuss this part. But there also is #15227 which further complicates things when considering two *moving* objects (although it should be possible to compute this similarly by using their relative positions and relative speeds, which should reduce this to the base case again).